compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -2076,16 +2076,40 @@ impl<'tcx> LifetimeContext<'_, 'tcx> {
                 for param in params {
                     if let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(param.span) {
                         if snippet.starts_with('&') && !snippet.starts_with("&'") {
-                            introduce_suggestion
-                                .push((param.span, format!("&'a {}", &snippet[1..])));
-                        } else if let Some(stripped) = snippet.strip_prefix("&'_ ") {
```

Comment:
there's a lot of repetition in this diff. maybe we should create a `(String, Span)` newtype with functions like `strip_prefix` or `slice` that operate on both at the same time?

Reply:
Done